src/databricks/labs/ucx/source_code/files.py

```diff
@@ -44,7 +50,114 @@ def __repr__(self):
         return f"<LocalFile {self._path}>"
 
 
-class LocalFileMigrator:
+class LocalDirectory(SourceContainer):
+    def __init__(self, path: Path, file_loader: FileLoader, dir_loader: DirectoryLoader):
+        self._path = path
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+
+    @property
+    def path(self) -> Path:
+        return self._path
+
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
+        return list(self._build_dependency_graph(parent))
+
+    def _build_dependency_graph(self, parent: DependencyGraph) -> Iterable[DependencyProblem]:
+        for child_path in self._path.iterdir():
+            if child_path.is_dir():
+                dependency = Dependency(self._dir_loader, child_path)
+                yield from parent.register_dependency(dependency).problems
+                continue
+            dependency = Dependency(self._file_loader, child_path)
+            yield from parent.register_dependency(dependency).problems
+
+    def __repr__(self):
+        return f"<LocalDirectory {self._path}>"
+
+
+# TODO remove duplicate code by inheriting from LocatedAdvice once https://github.com/databrickslabs/lsql/issues/103 is fixed
+# pylint: disable=duplicate-code
+@dataclass
+class FileProblem:
+    path: str
+    code: str
+    message: str
+    start_line: int
+    start_col: int
+    end_line: int
+    end_col: int
+
+
+class LocalFilesLinter:
```

Comment:
(re)use `databricks.labs.ucx.source_code.notebooks.sources.FileLinter`, because that implementation already has local notebook detection logic working.

Reply:
This one serves a different purpose, and uses the FileLinter, see line 130